packages/compiler/test/server/completion.ts

```diff
@@ -159,6 +159,159 @@ describe("compiler: server: completion", () => {
     );
   });
 
+  it("completes enum members", async () => {
+    const completions = await complete(
+      `
+      enum Fruit {
+        Orange,
+        Banana
+      }
+
+      model M {
+        f: Fruit.┆
+      }
+      `
+    );
+
+    check(
+      completions,
+      [
+        {
+          label: "Orange",
+          insertText: "Orange",
+          kind: CompletionItemKind.EnumMember,
+          documentation: undefined,
+        },
+        {
+          label: "Banana",
+          insertText: "Banana",
+          kind: CompletionItemKind.EnumMember,
+          documentation: undefined,
+        },
+      ],
+      {
+        allowAdditionalCompletions: false,
+      }
+    );
+  });
+
+  it("completes union variants", async () => {
+    const completions = await complete(
+      `
+      model Orange {}
+      model Banana {}
+      union Fruit {
+        orange: Orange,
+        banana: Banana
+      }
+
+      model M {
+        f: Fruit.┆
+      }
+      `
+    );
+
+    check(
+      completions,
+      [
+        {
+          label: "orange",
+          insertText: "orange",
+          kind: CompletionItemKind.EnumMember,
+          documentation: undefined,
+        },
+        {
+          label: "banana",
+          insertText: "banana",
+          kind: CompletionItemKind.EnumMember,
+          documentation: undefined,
+        },
+      ],
+      {
+        allowAdditionalCompletions: false,
+      }
+    );
+  });
+
+  it("completes namespace operations", async () => {
+    const completions = await complete(
+      `
+       namespace N {
+        op test(): void;
+       }
+       @dec(N.┆
```

Comment:
This breaks if you have a trailing `)` already, but it's an existing issue AFAICT. Filed #584